frame/contracts/src/storage.rs

```diff
@@ -230,8 +230,7 @@ where
 		let weight_per_key = (T::WeightInfo::on_initialize_per_trie_key(1) -
 			T::WeightInfo::on_initialize_per_trie_key(0))
 		.ref_time();
-		let decoding_weight =
-			weight_per_queue_item.scalar_saturating_mul(queue_len as RefTimeWeight);
+		let decoding_weight = weight_per_queue_item.saturating_mul(queue_len as RefTimeWeight);
```

Comment:
I don't think `queue_len` is a weight:

```suggestion
		let decoding_weight = weight_per_queue_item.saturating_mul(queue_len as u64);
```

Reply:
should be fixed here: https://github.com/paritytech/substrate/pull/12157